includes/class-wc-payments-apple-pay-registration.php

```diff
@@ -0,0 +1,335 @@
+			$error = __( 'Unable to copy domain association file to domain root.', 'woocommerce-payments' );
+		}
+
+		if ( isset( $error ) ) {
+			$url            = get_site_url() . '/' . $dir . '/' . $file;
+			$response       = @wp_remote_get( $url ); // @codingStandardsIgnoreLine
+			$already_hosted = @wp_remote_retrieve_body( $response ) === $new_contents; // @codingStandardsIgnoreLine
+			if ( ! $already_hosted ) {
+				Logger::log(
+					'Error: ' . $error . ' ' .
+					/* translators: expected domain association file URL */
+					sprintf( __( 'To enable Apple Pay, domain association file must be hosted at %s.', 'woocommerce-payments' ), $url )
+				);
+			}
+		}
+
+		Logger::log( __( 'Domain association file updated.', 'woocommerce-payments' ) );
+	}
+
+	/**
+	 * Adds a rewrite rule for serving the domain association file from the proper location.
+	 */
+	public function add_domain_association_rewrite_rule() {
+		$regex    = '^\.well-known\/apple-developer-merchantid-domain-association$';
+		$redirect = 'index.php?apple-developer-merchantid-domain-association=1';
+
+		add_rewrite_rule( $regex, $redirect, 'top' );
+	}
+
+	/**
+	 * Add to the list of publicly allowed query variables.
+	 *
+	 * @param  array $query_vars - provided public query vars.
+	 * @return array Updated public query vars.
+	 */
+	public function whitelist_domain_association_query_param( $query_vars ) {
+		$query_vars[] = 'apple-developer-merchantid-domain-association';
+		return $query_vars;
+	}
+
+	/**
+	 * Serve domain association file when proper query param is provided.
+	 *
+	 * @param object $wp WordPress environment object.
+	 */
+	public function parse_domain_association_request( $wp ) {
+		if (
+			! isset( $wp->query_vars['apple-developer-merchantid-domain-association'] ) ||
+			'1' !== $wp->query_vars['apple-developer-merchantid-domain-association']
+		) {
+			return;
+		}
+
+		$path = WCPAY_ABSPATH . '/apple-developer-merchantid-domain-association';
+		header( 'Content-Type: text/plain;charset=utf-8' );
+		echo esc_html( @file_get_contents( $path ) ); // @codingStandardsIgnoreLine
+		exit;
+	}
+
+	/**
+	 * Processes the Apple Pay domain verification.
+	 *
+	 * @return bool Whether domain verification succeeded.
+	 */
+	public function register_domain_with_apple() {
+		$error = null;
+		try {
+			$registration_response = $this->payments_api_client->register_domain_with_apple( $this->domain_name );
+
+			if ( isset( $registration_response['id'] ) ) {
+				$this->gateway_settings['apple_pay_verified_domain'] = $this->domain_name;
+				$this->gateway_settings['apple_pay_domain_set']      = 'yes';
+				update_option( 'woocommerce_woocommerce_payments_settings', $this->gateway_settings );
+
+				Logger::log( __( 'Your domain has been verified with Apple Pay!', 'woocommerce-payments' ) );
+
+				return true;
+			} elseif ( isset( $registration_response['error'] ) && isset( $registration_response['error']['message'] ) ) {
```

Comment:
The 2nd `isset` here should work and be enough on its own.

Reply:
That's true 😅 . Removed the first one.